src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -278,21 +278,24 @@ export class EthereumBlockchainService implements BlockchainService {
   }
 
   async _buildTransactionRequest(rootCid: CID): Promise<TransactionRequest> {
-    const rootStrHex = rootCid.toString(base16)
-    const hexEncoded = '0x' + (rootStrHex.length % 2 == 0 ? rootStrHex : '0' + rootStrHex)
-    logger.imp(`Hex encoded root CID ${hexEncoded}`)
-
     logger.debug('Preparing ethereum transaction')
     const baseNonce = await this.wallet.provider.getTransactionCount(this.wallet.address)
 
     if (!this.config.useSmartContractAnchors) {
+      const rootStrHex = rootCid.toString(base16)
+      const hexEncoded = '0x' + (rootStrHex.length % 2 == 0 ? rootStrHex : '0' + rootStrHex)
+      logger.imp(`Hex encoded root CID ${hexEncoded}`)
+
       return {
         to: this.wallet.address,
         data: hexEncoded,
         nonce: baseNonce,
       }
     }
-    const transactionRequest = await this._contract.populateTransaction.anchor(hexEncoded)
+
+    const transactionRequest = await this._contract.populateTransaction.anchorDagCbor(
+      rootCid.bytes.slice(4)
```

Comment:
Does this function accept a uint8array? If so nice, otherwise I guess it needs a hex encoded string?

Reply:
Nope doesn't seem to. I'll use the hex encoded string